src/main/java/db/UsersDao.java

```diff
@@ -0,0 +1,28 @@
+package db;
+import db.Dbinitializer;
+import sda.java.loginproject.entity.Users;
+
+public class UsersDao extends Dbinitializer {
+
+    public void insertUser (Users user) {
+        openSessionAndTransaction();
+        session.persist(user);
+        user.setUsername("razvanchitac");
+        closeSessionAndTransaction();
+    }
+
+    public Users findUser(String username) {
+        openSessionAndTransaction();
+        Users user = session.find(Users.class, username);
+        System.out.println(user);
+        closeSessionAndTransaction();
+        return user;
+    }
+
+    public void updateUser (Users user) {
```

Comment:
does this user always contain the id when updating?

Reply:
Yes, id is unique, right ?

Reply:
When you update a user, the id will always need to be provided there, otherwise it will create a new raw in the database. You should check first to see if the id is set on the user, or check it in a logic above this method call.

Reply:
ok acum ?